### PR TITLE
add canmodifyconstraint

### LINKED ISSUE #85: Check if constraints can be modified

### PR DESCRIPTION
**Pull request by joaquimg**
closes #85

### CODE REVIEW

**Comment by joaquimg** [comment]
Should we have a fallback like in https://github.com/JuliaOpt/MathOptInterface.jl/pull/108#issuecomment-321199454

**Comment by blegat** [comment]
We can have a fallback for AbstractFunctionModification but it needs to be in MOIU since function modification is implemented there

File: src/constraints.jl [hunk 1]
```diff
@@ -19,7 +19,51 @@ function addconstraints! end
 # default fallback
 addconstraints!(m::AbstractSolverInstance, funcs, sets) = addconstraint!.(m, funcs, sets)
 
-# TODO: method to query if solver supports this type of modification
+"""
+## Modify Function
+
+    canmodifyconstraint(m::AbstractSolverInstance, c::ConstraintReference, func::F)::Bool
```
Comment (odow) [inline src/constraints.jl]: `c::ConstraintReference{F,S}, func::F)::Bool`

File: src/constraints.jl [hunk 2]
```diff
@@ -19,7 +19,51 @@ function addconstraints! end
...
+If `c` is a `ConstraintReference{ScalarAffineFunction,S}` and `v1` and `v2` are `VariableReference` objects,
+
+```julia
+canmodifyconstraint(m, c, ScalarAffineFunction([v1,v2],[1.0,2.0],5.0))
+canmodifyconstraint(m, c, SingleVariable(v1)) # false
+```
+
+## Modify Set
+
+    canmodifyconstraint(m::AbstractSolverInstance, c::ConstraintReference, S::S)::Bool
```
Comment (odow) [inline src/constraints.jl]: `c::ConstraintReference{F,S}, set::S)::Bool`

**Comment by odow** [comment]
> Should we have a fallback like in #108 (comment) ?

No. I think these methods should stay as simple as possible. Modification means in-place.

**Pull request merged**